### DISCUSSION TITLE
QUESTION: insert_image cover the origin text

### DISCUSSION THREAD

**Discussion by Syntamin** [pymupdf/PyMuPDF#3717] [Q&A] [answered]
### Description of the bug

Dear [JorjMcKie](https://github.com/JorjMcKie),

I am currently working with a PDF document that contains text, and I have encountered a challenge. When I use the `page.insert_image` function to insert an image into the PDF, the new image overlays the existing text. However, my goal is to have the image appear beneath the text, ensuring the text remains visible.

I have searched the documentation but have not found a method to detect when text is covered by an image or to adjust the layering order during the insertion process. Could you please advise if there is a way to achieve this, or if there is any functionality in the library that I might have overlooked?

Thank you for your time and assistance.

### How to reproduce the bug

```
for rect_index, rect in enumerate(image_rect, start=0):
    print(rect_index, rect)

    image_xref = page.insert_image(
        image_rect[rect_index],
        stream=compressed_base_image_bytes,
        mask=compressed_mask_image_bytes,
        xref=image_xref,
    )

    page.apply_redactions(pymupdf.PDF_REDACT_IMAGE_NONE)
```
![cover_text](https://github.com/user-attachments/assets/5aa03d35-5801-4835-9170-d92be37433ae)

### PyMuPDF version

1.24.7

### Operating system

MacOS

### Python version

3.11

**Comment by JorjMcKie** [comment]
This is a typical Discussions item .... transferring first.

**Comment by JorjMcKie** [comment, marked as answer]
Simply add parameter `overlay=False`:
```python
for rect_index, rect in enumerate(image_rect, start=0):
    print(rect_index, rect)

    image_xref = page.insert_image(
        image_rect[rect_index],
        stream=compressed_base_image_bytes,
        mask=compressed_mask_image_bytes,
        xref=image_xref,
        overlay=False,  # <=== puts in background
    )
    # you do not need this:
    # page.apply_redactions(pymupdf.PDF_REDACT_IMAGE_NONE)
``

**Reply by Syntamin** [reply]
Thank you, that is indeed the case.👍
